checker/jdk/nullness/src/org/w3c/dom/Document.java

```diff
@@ -0,0 +1,847 @@
+ * <p>See also the <a href='http://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407'>Document Object Model (DOM) Level 3 Core Specification</a>.
+ */
+public interface Document extends Node {
+    /**
+     * The Document Type Declaration (see <code>DocumentType</code>)
+     * associated with this document. For XML documents without a document
+     * type declaration this returns <code>null</code>. For HTML documents,
+     * a <code>DocumentType</code> object may be returned, independently of
+     * the presence or absence of document type declaration in the HTML
+     * document.
+     * <br>This provides direct access to the <code>DocumentType</code> node,
+     * child node of this <code>Document</code>. This node can be set at
+     * document creation time and later changed through the use of child
+     * nodes manipulation methods, such as <code>Node.insertBefore</code>,
+     * or <code>Node.replaceChild</code>. Note, however, that while some
+     * implementations may instantiate different types of
+     * <code>Document</code> objects supporting additional features than the
+     * "Core", such as "HTML" [<a href='http://www.w3.org/TR/2003/REC-DOM-Level-2-HTML-20030109'>DOM Level 2 HTML</a>]
+     * , based on the <code>DocumentType</code> specified at creation time,
+     * changing it afterwards is very unlikely to result in a change of the
+     * features supported.
+     *
+     * @since DOM Level 3
+     */
+    @Pure public @Nullable DocumentType getDoctype();
+
+    /**
+     * The <code>DOMImplementation</code> object that handles this document. A
+     * DOM application may use objects from multiple implementations.
+     */
+    @Pure public DOMImplementation getImplementation();
+
+    /**
+     * This is a convenience attribute that allows direct access to the child
+     * node that is the document element of the document.
+     */
+    @Pure public Element getDocumentElement();
+
+    /**
+     * Creates an element of the type specified. Note that the instance
+     * returned implements the <code>Element</code> interface, so attributes
+     * can be specified directly on the returned object.
+     * <br>In addition, if there are known attributes with default values,
+     * <code>Attr</code> nodes representing them are automatically created
+     * and attached to the element.
+     * <br>To create an element with a qualified name and namespace URI, use
+     * the <code>createElementNS</code> method.
+     * @param tagName The name of the element type to instantiate. For XML,
+     *   this is case-sensitive, otherwise it depends on the
+     *   case-sensitivity of the markup language in use. In that case, the
+     *   name is mapped to the canonical form of that markup by the DOM
+     *   implementation.
+     * @return A new <code>Element</code> object with the
+     *   <code>nodeName</code> attribute set to <code>tagName</code>, and
+     *   <code>localName</code>, <code>prefix</code>, and
```

Comment:
This documentation, and nearly-identical documentation in `createAttribute`, suggests that all of `localName`, `prefix`, and `namespaceURI` should be `@Nullable` throughout unless there is an indication otherwise.  Are there such indications?